app/src/main/java/com/amaze/filemanager/ui/fragments/FtpServerFragment.kt

```diff
@@ -449,28 +460,40 @@ class FtpServerFragment : Fragment(R.layout.fragment_ftp) {
                             .positiveColor(accentColor)
                             .negativeText(R.string.cancel)
                             .negativeColor(accentColor)
-                            .onPositive { dialog, _ ->
-                                activityResultHandlerOnFtpServerPathGrantedSafAccess.launch(
-                                    Intent(Intent.ACTION_OPEN_DOCUMENT_TREE).also {
-                                        if (SDK_INT >= O &&
-                                            directoryUri.startsWith(defaultPathFromPreferences)
-                                        ) {
-                                            it.putExtra(
-                                                EXTRA_INITIAL_URI,
-                                                DocumentsContract.buildDocumentUri(
-                                                    "com.android.externalstorage.documents",
-                                                    "primary:" +
-                                                        directoryUri
-                                                            .substringAfter(
-                                                                defaultPathFromPreferences
-                                                            )
+                            .onPositive(fun(dialog: MaterialDialog, _: DialogAction) {
+                                val intent = Intent(Intent.ACTION_OPEN_DOCUMENT_TREE)
+
+                                if (intent.resolveActivity(mainActivity.packageManager) != null) {
```

Comment:
Can you create a kotlin extension function on `Intent` instead of writing if else conditions everywhere please? Take callbacks as args.